src/ClassReflectionParser.php

```diff
@@ -19,7 +20,7 @@
     public function __construct()
     {
         $parserFactory = new ParserFactory();
-        $this->phpParser = $parserFactory->create(ParserFactory::PREFER_PHP7);
+        $this->phpParser = $parserFactory->createForVersion(PhpVersion::fromString('7.0'));
```

Comment:
Hi,

I started to look into this yesterday but now I see work is already in progress. Most of my  changes are already included in this and the other already merged PR. However  I think it would make more sense to use the new `createForHostVersion` method here.

Reply:
I went with what was used in rector. what would be the benefit of using the method you suggested?

Reply:
Based on comments in [ParserFactory](https://github.com/nikic/PHP-Parser/blob/master/lib/PhpParser/ParserFactory.php) the createForVersion makes a lexer that tries to parse php code that is now considered invalid while tryinng too support the latest language constructs also.

In PhpParser 4 there was no PHP8 parser, it used the PHP7 parser for that too but setting it to 7.0 here means it will never use the new PHP8 parser.

Reply:
I am not sure what we need for this particular project here. @TomasVotruba do you have an opinion on that?

Reply:
An even better solution would be to import the parser from PHPstan itself because it has a factory for it but I never tried it myself.

As far as I can see it can be injected to the constructor:
- factory: https://github.com/phpstan/phpstan-src/blob/2.0.x/src/Parser/PhpParserFactory.php
- usage: https://github.com/phpstan/phpstan-src/blob/2.0.x/src/Type/ClosureTypeFactory.php

It would make sense to use the same parser as the rest of the system.

Reply:
I think host would be the best version, to fit user's code.

Reply:
adjusted with the lastest commit